File: Level1/Lessons12931/gamjapark.cpp
// 자릿수 더하기

#include <vector>
#include <string>

using namespace std;
int solution(int n)
{
    int answer = 0;
    
    string intString = to_string(n);
    int Size = intString.size();
    const char* stringOneChar = intString.c_str();
    
    for(int i = 0; i < Size; i++){
        answer += (stringOneChar[i] - '0');
    }
    return answer;
}

/*
정확성  테스트
테스트 1 〉	통과 (0.01ms, 3.97MB)
테스트 2 〉	통과 (0.01ms, 3.97MB)
테스트 3 〉	통과 (0.02ms, 3.95MB)
테스트 4 〉	통과 (0.01ms, 3.95MB)
테스트 5 〉	통과 (0.02ms, 3.97MB)
테스트 6 〉	통과 (0.01ms, 3.95MB)
테스트 7 〉	통과 (0.02ms, 3.96MB)
테스트 8 〉	통과 (0.01ms, 3.95MB)
테스트 9 〉	통과 (0.01ms, 3.89MB)
테스트 10 〉	통과 (0.01ms, 3.88MB)
테스트 11 〉	통과 (0.02ms, 3.93MB)
테스트 12 〉	통과 (0.01ms, 3.96MB)
테스트 13 〉	통과 (0.01ms, 3.83MB)
테스트 14 〉	통과 (0.01ms, 3.93MB)
테스트 15 〉	통과 (0.02ms, 3.89MB)
테스트 16 〉	통과 (0.01ms, 3.97MB)
테스트 17 〉	통과 (0.01ms, 3.95MB)
테스트 18 〉	통과 (0.01ms, 3.97MB)
테스트 19 〉	통과 (0.02ms, 3.97MB)
테스트 20 〉	통과 (0.01ms, 3.93MB)
테스트 21 〉	통과 (0.02ms, 3.94MB)
채점 결과
정확성: 100.0
합계: 100.0 / 100.0
*/

File: Level2/Lessons12924/gamjapark.py
# 숫자의 표현

def solution(n):
    answer = 0
    numSum = 0
    sNum = 1
    for i in range(1, n+1):
        numSum += i
        while numSum > n:
            numSum -= sNum
            sNum += 1
                
        if numSum == n:
            answer += 1
    return answer
    
'''
채점을 시작합니다.
정확성  테스트
테스트 1 〉	통과 (0.01ms, 10.1MB)
테스트 2 〉	통과 (0.13ms, 10.1MB)
테스트 3 〉	통과 (0.11ms, 10.1MB)
테스트 4 〉	통과 (0.11ms, 10.2MB)
테스트 5 〉	통과 (0.04ms, 10.1MB)
테스트 6 〉	통과 (0.01ms, 10.2MB)
테스트 7 〉	통과 (0.10ms, 10.2MB)
테스트 8 〉	통과 (0.05ms, 10.2MB)
테스트 9 〉	통과 (0.01ms, 10.2MB)
테스트 10 〉	통과 (0.20ms, 10.1MB)
테스트 11 〉	통과 (0.17ms, 10.1MB)
테스트 12 〉	통과 (0.11ms, 10.1MB)
테스트 13 〉	통과 (0.12ms, 10.1MB)
테스트 14 〉	통과 (0.09ms, 10.2MB)
효율성  테스트
테스트 1 〉	통과 (1.63ms, 10.1MB)
테스트 2 〉	통과 (1.33ms, 10.2MB)
테스트 3 〉	통과 (1.42ms, 10.1MB)
테스트 4 〉	통과 (1.33ms, 10.2MB)
테스트 5 〉	통과 (1.40ms, 10.2MB)
테스트 6 〉	통과 (1.40ms, 10.2MB)
채점 결과
정확성: 70.0
효율성: 30.0
합계: 100.0 / 100.0
'''